File: backend/api/mecanicsync/src/main/java/com/example/mecanicsync/entities/Vehiculo.java
package com.example.mecanicsync.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name="vehiculos")
public class Vehiculo {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_vehiculo")
    private Integer id;

    @ManyToOne
    @JoinColumn(name = "id_cliente")
    private Cliente cliente;

    @ManyToOne
    @JoinColumn(name = "id_modelo")
    private Modelo modelo;

    private String matricula;

    @Column(name = "anio_matriculacion")
    private Integer anioMatriculacion;

    @OneToMany(mappedBy = "vehiculo", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Reparacion> reparaciones;

}

File: android_app/MecanicSyncApp/MecanicSync/app/src/main/java/com/example/mecanicsync/vehiculos/ui/model/Vehiculo.kt
package com.example.mecanicsync.vehiculos.ui.model

import com.example.mecanicsync.reparaciones.ui.model.Reparacion

data class Vehiculo(
    val id_vehiculo: Int = 0,
    val id_cliente: Int = 0,
   val id_modelo: Int = 0,
   val matricula: String= "",
    val marca: String= "",
    val modelo: String= "",
   val anioMatriculacion: Int= 0,
   val reparaciones: List<Reparacion> = emptyList()
)
